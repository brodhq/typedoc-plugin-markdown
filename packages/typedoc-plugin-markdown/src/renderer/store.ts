import { ProjectReflection } from 'typedoc';

import { PluginOptions, State } from './types';

const state: State = {
  project: new ProjectReflection(''),
  currentModel: null,
  hasReadme: null,
  currentUrl: '',
  projectName: '',
  globalsName: '',
  projectUrl: '',
  options: {},
};

export const useState = () => state;

export function setDefaultState(
  project: ProjectReflection,
  options: Partial<PluginOptions>,
) {
  setState({
    projectName: project.name,
    projectUrl: project.url,
    project: project,
    globalsName:
      options.entryPoints && options.entryPoints.length > 1
        ? 'Modules'
        : 'Exports',
    options: options,
    hasReadme: options.readme !== 'none',
  });
}

export function setState(nextState) {
  Object.assign(state, nextState);
}
