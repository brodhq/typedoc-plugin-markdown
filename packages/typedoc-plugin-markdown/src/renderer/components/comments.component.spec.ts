import { TestApp } from '../../../test/test-app';
import { setState } from '../store';
import { CommentsComponent } from './comments.component';

describe(`Comments Component`, () => {
  let testApp: TestApp;

  beforeAll(async () => {
    testApp = new TestApp(['comments.ts', 'variables.ts']);
    await testApp.bootstrap();
  });

  test(`should build @link reference'`, () => {
    const page = testApp.getPageByModelName('comments').model;
    setState({ currentModel: page });
    const reflection = testApp.project.findReflectionByName(
      'commentsWithDocLinks',
    );
    expect(CommentsComponent(reflection.comment)).toMatchSnapshot();
  });

  test(`should convert symbols brackets to symbol links'`, () => {
    const page = testApp.getPageByModelName('comments').model;
    setState({ currentModel: page });
    const reflection = testApp.project.findReflectionByName(
      'commentsWithSymbolLinks',
    );
    expect(CommentsComponent(reflection.comment)).toMatchSnapshot();
  });

  test(`should convert comments with tags'`, () => {
    const page = testApp.getPageByModelName('comments').model;
    setState({ currentModel: page });
    const reflection = testApp.project.findReflectionByName('commentsWithTags');
    expect(CommentsComponent(reflection.comment)).toMatchSnapshot();
  });

  test(`should allow html in comments'`, () => {
    const page = testApp.getPageByModelName('comments').model;
    setState({ currentModel: page });
    const reflection = testApp.project.findReflectionByName('commentsWithHTML');
    expect(CommentsComponent(reflection.comment)).toMatchSnapshot();
  });
});
