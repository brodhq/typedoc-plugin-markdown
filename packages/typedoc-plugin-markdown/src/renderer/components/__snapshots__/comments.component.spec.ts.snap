// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments Component should allow html in comments' 1`] = `
"<p>
You can write <strong>HTML</strong> tags directly in <a href=\\"comments\\">comments</a>
</p>
<ul>
<li>List item</li>
</ul>"
`;

exports[`Comments Component should build @link reference' 1`] = `
"- [\`CommentClass\`](../classes/comments.commentclass.md)
- [CommentClass.comment](../classes/comments.commentclass.md#comment)
- {@link unknownA}
- [commentsInReturn](comments.md#commentsinreturn)
- [Google](https://www.google.com)
- [GitHub](https://github.com)"
`;

exports[`Comments Component should convert comments with tags' 1`] = `
"**\`name\`** Tag description on same line

**\`description\`**
Tag description on new line

- Tag description on another line


**\`deprecated\`**
Another tag description
"
`;

exports[`Comments Component should convert symbols brackets to symbol links' 1`] = `
"Additionally you can link to other classes, members or functions using double square brackets.

- [CommentClass](../classes/comments.commentclass.md)
- [commentsInReturn](comments.md#commentsinreturn)
- [[unknownB]]
"
`;
